components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/internal/IdentityUserProfileServiceComponent.java

```diff
@@ -73,6 +74,17 @@ protected void activate(ComponentContext ctxt) {
             } else {
                 log.error("User profile management - ProfileMgtEventListener could not be registered.");
             }
+
+            ServiceRegistration userProfileAdminSR =
```

Comment:
Exposing an admin service as an OSGi layer directly doesn't sound right. Can we introduce the separate OSGi service?
You can use the same OSGi service within the admin service as well.